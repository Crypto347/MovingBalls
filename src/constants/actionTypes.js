export const MOVE_CIRCLE_X_COORDINATE = 'MOVE_CIRCLE_X_COORDINATE';
export const MOVE_CIRCLE_Y_COORDINATE = 'MOVE_CIRCLE_Y_COORDINATE';
export const CHANGE_DIRECTION_OF_X_MOVE = 'CHANGE_DIRECTION_OF_X_MOVE';
export const CHANGE_DIRECTION_OF_Y_MOVE = 'CHANGE_DIRECTION_OF_Y_MOVE';
export const UPDATE_COORDINATES = 'UPDATE_COORDINATES';
export const UPDATE_VELOCITIES = 'UPDATE_VELOCITIES';
export const FILL_CIRCLES_ARRAY = 'FILL_CIRCLES_ARRAY';
export const UPDATE_CIRCLES_ARRAY = 'UPDATE_CIRCLES_ARRAY';
export const GET_NUMBERS_OF_BALLS = 'GET_NUMBERS_OF_BALLS';
