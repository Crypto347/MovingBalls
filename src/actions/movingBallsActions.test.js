/**
* Libraries
*/

import configureStore from 'redux-mock-store';

/**
* Constants
*/

import * as Actions from '.';
import * as actionTypes from "../constants/actionTypes";

/**
* Tests
*/

const mockStore = configureStore();
const store = mockStore();

describe('movingBallsActions', () => {
    beforeEach(() => { // Runs before each test in the suite
      store.clearActions();
    });

    it('Dispatches the correct action and payload (moveCircleXCoordinate)', () => {
        const expectedActions = [
            {
                type: actionTypes.MOVE_CIRCLE_X_COORDINATE,
                dx: 467,
                id: 3
            }
        ];

        store.dispatch(Actions.moveCircleXCoordinate(467, 3));
        expect(store.getActions()).toEqual(expectedActions);
        expect(store.getActions()).toMatchSnapshot();
    });
});
