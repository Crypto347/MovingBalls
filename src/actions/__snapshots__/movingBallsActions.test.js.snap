// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`movingBallsActions Dispatches the correct action and payload (moveCircleXCoordinate) 1`] = `
Array [
  Object {
    "dx": 467,
    "id": 3,
    "type": "MOVE_CIRCLE_X_COORDINATE",
  },
]
`;
